src/components/attendance/attendance.repository.ts

```diff
@@ -0,0 +1,70 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { AttendanceEntity } from './entity/attendance.entity';
+import { Repository } from 'typeorm';
+import { StudentEntity } from '../student/entity/student.entity';
+import { Injectable, NotFoundException } from '@nestjs/common';
+
+@Injectable()
+export class AttendanceRepository {
+  constructor(
+    @InjectRepository(AttendanceEntity)
+    private attendanceRep: Repository<AttendanceEntity>,
+    @InjectRepository(StudentEntity)
+    private studentRep: Repository<StudentEntity>,
+  ) {}
+
+  public async getAllAttendances(): Promise<AttendanceEntity[]> {
+    return await this.attendanceRep.find();
+  }
+
+  public async getAttendanceById(id: string): Promise<AttendanceEntity> {
+    const attendance: AttendanceEntity  = await this.attendanceRep.findOne({ where: { id: id } });
+    if(!attendance){
+        throw new NotFoundException({message:`please Enter valid id`})
+    }
+    return attendance;
+  }
+
+  public async getStudentById(id: string) {
+    return await this.studentRep.findOne({ where: { id: id } });
+  }
+
+  public async createAttendance(
+    attendance,
+    StudentEntity,
+  ): Promise<AttendanceEntity> {
+    const AttendanceEntity = await this.attendanceRep.create({
+      isPresent: attendance.isPresent,
+      student: StudentEntity,
+    });
+    return await this.attendanceRep.save(AttendanceEntity);
+  }
+
+  public async updateAttendance(
+    attendance,
+    id: string,
+  ): Promise<AttendanceEntity[]> {
+    const existingAttendance: AttendanceEntity =
+      await this.attendanceRep.findOne({ where: { id: id } });
+    const updatedResult: AttendanceEntity = await this.attendanceRep.merge(
```

Comment:
my mentor told be to go for db based query operations instead of using object.assign and .merge comes with Typeorm so it is better to use that one in old node project i had used Object.assign only but then i updated with mongoose update query